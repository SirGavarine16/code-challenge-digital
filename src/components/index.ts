export { default as WelcomeHeader } from './WelcomeHeader';
export { default as SectionLabel } from './SectionLabel';
export { default as PointsDisplay } from './PointsDisplay';
export { default as MovementsList } from './MovementsList';
export { default as MovementCard } from './MovementCard';
export { default as MovementListItemSeparator } from './MovementListItemSeparator';
export { default as Button } from './Button';
export { default as MovementHeader } from './MovementHeader';
export { default as MovementProductImage } from './MovementProductImage';
export { default as EmptyMovementsList } from './EmptyMovementsList';