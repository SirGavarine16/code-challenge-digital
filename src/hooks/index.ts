export { default as useProductsFetch } from './useProductsFetch';
export { default as useProductPoints } from './useProductPoints';