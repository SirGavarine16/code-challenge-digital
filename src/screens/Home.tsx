import React, { FC } from 'react';
import { Text, View } from 'react-native';

interface Props {
}

const Home: FC<Props> = () => {
    return (
        <View>
            <Text>Home Screen</Text>
        </View>
    );
};

export default Home;
