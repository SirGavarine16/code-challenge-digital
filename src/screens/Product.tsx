import React, { FC } from 'react';
import { Text, View } from 'react-native';

interface Props {
}

const Product: FC<Props> = () => {
    return (
        <View>
            <Text>Product Screen</Text>
        </View>
    );
};

export default Product;
