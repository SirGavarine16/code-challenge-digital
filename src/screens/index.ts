export { default as Home } from './Home';
export { default as Movement } from './Movement';