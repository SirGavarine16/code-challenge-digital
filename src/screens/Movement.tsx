import React, { FC, useEffect } from 'react';
import { Animated, StyleSheet, Text, View } from 'react-native';
import { useSafeAreaInsets } from 'react-native-safe-area-context';
import { StackScreenProps } from '@react-navigation/stack';

import { Button, MovementHeader, MovementProductImage } from '../components';
import { AppStackParams } from '../navigation/AppStack';
import { colors, spacing } from '../themes/AppTheme';
import { formatDate } from '../utils/formats';
import { useAnimations } from '../hooks';

interface Props extends StackScreenProps<AppStackParams, 'MovementScreen'> {
}

const Movement: FC<Props> = ({ route, navigation }) => {
    const { product, image, createdAt, is_redemption, points } = route.params;
    const { container, contentContainer, dataContainer, descriptionLabel, dateLabel, pointsLabel } = styles;

    const { top, bottom } = useSafeAreaInsets();
    const { fadeIn, fadeOut, movePosition, opacity, position } = useAnimations();

    useEffect(() => {
        fadeIn();
        movePosition(100, 600);
        return () => {
            fadeOut();
        }
    }, []);

    return (
        <View style={[container, { paddingBottom: bottom === 0 ? 20 : bottom }]}>
            <MovementHeader product={product} top={top} />
            <View style={contentContainer}>
                <MovementProductImage image={image} />
                <View style={dataContainer}>
                    <Text testID='product-details' style={descriptionLabel}>
                        Detalles del producto:
                    </Text>
                    <Text testID='sale-date' style={dateLabel}>
                        {`Comprado el ${formatDate(createdAt)}`}
                    </Text>
                    <Text testID='points-details' style={descriptionLabel}>
                        Con esta compra {is_redemption ? 'canjeaste' : 'acumulaste'}:
                    </Text>
                    <Text testID='points' style={pointsLabel}>
                        {points} puntos
                    </Text>
                </View>
                <Animated.View style={{ opacity, transform: [{ translateX: position }] }}>
                    <Button title='Aceptar' onPress={() => navigation.navigate('HomeScreen')} />
                </Animated.View>
            </View>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    contentContainer: {
        flex: 1,
        paddingHorizontal: spacing.paddingHorizontal,
    },
    dataContainer: {
        marginTop: 32,
        flex: 1,
    },
    descriptionLabel: {
        color: colors.textSecondary,
        fontSize: 14,
        fontWeight: '800',
    },
    dateLabel: {
        fontSize: 16,
        fontWeight: '800',
        marginVertical: 20,
    },
    pointsLabel: {
        fontSize: 24,
        fontWeight: '800',
        marginVertical: 32,
    }
})

export default Movement;
