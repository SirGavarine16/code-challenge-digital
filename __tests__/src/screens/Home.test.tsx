import { render } from '@testing-library/react-native';

import { Home } from '../../../src/screens';

describe('<Home />', () => {
    it('should render correctly', () => {
        const screen = render(<Home />);
        expect(screen).toMatchSnapshot();
    });
});