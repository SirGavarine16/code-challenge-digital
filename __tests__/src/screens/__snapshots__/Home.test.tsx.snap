// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render correctly 1`] = `
<View>
  <Text>
    Home Screen
  </Text>
</View>
`;
